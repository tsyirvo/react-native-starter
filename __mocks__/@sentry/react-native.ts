module.exports = {
  ReactNavigationInstrumentation: jest.fn().mockImplementation(() => {}),
};
