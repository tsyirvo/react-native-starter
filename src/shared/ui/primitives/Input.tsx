/* eslint-disable react/jsx-props-no-spreading */

import { forwardRef } from 'react';
import { TextInput, TextInputProps } from 'react-native';

import { makeAppStyles, theme, fontSizes } from '$core/theme';

import { Box } from './Box';
import { Text } from './Text';

type Props = TextInputProps & {
  label?: string;
  error?: string;
  isEditable?: boolean;
};

const useStyles = makeAppStyles(({ colors }) => ({
  input: {
    fontSize: fontSizes.regular,
    borderBottomWidth: 1,
    padding: 0,
    paddingBottom: 5,
  },
  defaultState: {
    borderBottomColor: colors.grey,
  },
  errorState: {
    borderBottomColor: colors.red,
  },
}));

export const Input = forwardRef<TextInput, Props>(
  ({ label, error, isEditable = true, ...props }, ref) => {
    const styles = useStyles();

    const errorStyles = error ? styles.errorState : styles.defaultState;

    return (
      <Box width="100%">
        {label && (
          <Text color="black" mb="global_8" testID="inputID-label">
            {label}
          </Text>
        )}

        <TextInput
          ref={ref}
          editable={isEditable}
          placeholderTextColor={theme.colors.grey}
          style={[styles.input, errorStyles]}
          testID="inputID"
          underlineColorAndroid="transparent"
          {...props}
          onChangeText={props.onChangeText}
        />

        {!!error && (
          <Box alignItems="center" flexDirection="row" testID="inputID-error">
            <Text color="red" variant="small">
              {error}
            </Text>
          </Box>
        )}
      </Box>
    );
  },
);
