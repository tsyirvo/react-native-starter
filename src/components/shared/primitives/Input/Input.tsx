/* eslint-disable react/jsx-props-no-spreading */

import { forwardRef } from 'react';
import { TextInput, TextInputProps } from 'react-native';

import { Box, Text } from '$components/shared/primitives';
import { fontSizes } from '$styles/fonts';
import { makeAppStyles, theme } from '$styles/theme';

type Props = TextInputProps & {
  label?: string;
  error?: string;
  isEditable?: boolean;
};

const useStyles = makeAppStyles(({ colors }) => ({
  input: {
    fontSize: fontSizes.regular,
    borderBottomWidth: 1,
    padding: 0,
    paddingBottom: 5,
  },
  defaultState: {
    borderBottomColor: colors.grey,
  },
  errorState: {
    borderBottomColor: colors.red,
  },
}));

const Input = forwardRef<TextInput, Props>(
  ({ label, error, isEditable = true, ...props }, ref) => {
    const styles = useStyles();

    const errorStyles = error ? styles.errorState : styles.defaultState;

    return (
      <Box width="100%">
        {label && (
          <Text color="black" mb="global_8" testID="input-label">
            {label}
          </Text>
        )}

        <TextInput
          ref={ref}
          style={[styles.input, errorStyles]}
          underlineColorAndroid="transparent"
          placeholderTextColor={theme.colors.grey}
          editable={isEditable}
          {...props}
        />

        {!!error && (
          <Box flexDirection="row" alignItems="center" testID="input-error">
            <Text variant="small" color="red">
              {error}
            </Text>
          </Box>
        )}
      </Box>
    );
  },
);

export default Input;
