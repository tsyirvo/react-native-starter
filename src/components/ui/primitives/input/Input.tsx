/* eslint-disable react/jsx-props-no-spreading */

import { forwardRef } from 'react';
import { TextInput, TextInputProps } from 'react-native';

import { Box, Text } from '$components/ui/primitives';
import { fontSizes } from '$styles/fonts';
import { makeAppStyles, theme } from '$styles/theme';

type Props = TextInputProps & {
  label?: string;
  error?: string;
  isEditable?: boolean;
};

const useStyles = makeAppStyles(({ colors }) => ({
  input: {
    fontSize: fontSizes.regular,
    borderBottomWidth: 1,
    padding: 0,
    paddingBottom: 5,
  },
  defaultState: {
    borderBottomColor: colors.grey,
  },
  errorState: {
    borderBottomColor: colors.red,
  },
}));

const Input = forwardRef<TextInput, Props>(
  ({ label, error, isEditable = true, ...props }, ref) => {
    const styles = useStyles();

    const errorStyles = error ? styles.errorState : styles.defaultState;

    return (
      <Box width="100%">
        {label && (
          <Text color="black" mb="global_8" testID="inputID-label">
            {label}
          </Text>
        )}

        <TextInput
          ref={ref}
          editable={isEditable}
          placeholderTextColor={theme.colors.grey}
          style={[styles.input, errorStyles]}
          testID="inputID"
          underlineColorAndroid="transparent"
          {...props}
          onChangeText={props.onChangeText}
        />

        {!!error && (
          <Box alignItems="center" flexDirection="row" testID="inputID-error">
            <Text color="red" variant="small">
              {error}
            </Text>
          </Box>
        )}
      </Box>
    );
  },
);

export default Input;
