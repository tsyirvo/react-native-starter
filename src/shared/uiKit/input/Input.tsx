/* eslint-disable react/jsx-props-no-spreading */

import { forwardRef } from 'react';
import { TextInput } from 'react-native';
import type { TextInputProps } from 'react-native';

import { makeAppStyles, theme, fontSizes } from '$core/theme';

import { Box, Text } from '../primitives';

export interface InputProps extends TextInputProps {
  label?: string;
  error?: string;
  isEditable?: boolean;
}

export const Input = forwardRef<TextInput, InputProps>(
  ({ label, error, isEditable = true, ...props }, ref) => {
    const styles = useStyles();

    const errorStyles = error ? styles.errorState : styles.defaultState;

    return (
      <Box width="100%">
        {label ? (
          <Text color="black" mb="spacing_8" testID="inputID-label">
            {label}
          </Text>
        ) : null}

        <TextInput
          ref={ref}
          editable={isEditable}
          placeholderTextColor={theme.colors.secondary_100}
          style={[styles.input, errorStyles]}
          testID="inputID"
          underlineColorAndroid="transparent"
          onChangeText={props.onChangeText}
          {...props}
        />

        {!!error && (
          <Box alignItems="center" flexDirection="row" testID="inputID-error">
            <Text color="red" variant="small">
              {error}
            </Text>
          </Box>
        )}
      </Box>
    );
  },
);

const useStyles = makeAppStyles(({ colors }) => ({
  input: {
    fontSize: fontSizes.regular,
    borderBottomWidth: 1,
    padding: 0,
    paddingBottom: 5,
  },
  defaultState: {
    borderBottomColor: colors.secondary_80,
  },
  errorState: {
    borderBottomColor: colors.red,
  },
}));

Input.displayName = 'Input';
