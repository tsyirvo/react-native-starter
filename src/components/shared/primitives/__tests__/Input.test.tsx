/* eslint-disable react/jsx-props-no-spreading */

import { fireEvent } from '@testing-library/react-native';
import React from 'react';

import render from '$tests/utils';

import Input from '../Input';

describe('Shared primitives Input component', () => {
  // given
  const props = {
    onChangeText: jest.fn(),
    placeholder: 'Placeholder',
  };

  it('should render correctly', () => {
    // When
    const wrapper = render(<Input {...props} />);

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger the onChangeText props when typing', () => {
    // When
    const { getByPlaceholderText } = render(<Input {...props} />);

    const input = getByPlaceholderText('Placeholder');

    fireEvent.changeText(input, 'data');

    // Then
    expect(props.onChangeText).toHaveBeenCalled();
  });

  it('should not show error by default', () => {
    // When
    const { queryByTestId } = render(<Input {...props} />);

    const error = queryByTestId('input-error');

    // Then
    expect(error).toBeNull();
  });

  it('should show error when defined', () => {
    // When
    const { getByTestId } = render(<Input {...props} error="Some error" />);

    const error = getByTestId('input-error');

    // Then
    expect(error).toBeDefined();
  });

  it('should not show a label if not provided', () => {
    // When
    const { queryByTestId } = render(<Input {...props} />);

    const label = queryByTestId('input-label');

    // Then
    expect(label).toBeNull();
  });

  it('should show a label when defined', () => {
    // When
    const { getByTestId } = render(<Input {...props} label="Some label" />);

    const label = getByTestId('input-label');

    // Then
    expect(label).toBeDefined();
  });
});
