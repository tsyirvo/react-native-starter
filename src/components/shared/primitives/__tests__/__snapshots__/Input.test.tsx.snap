// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared primitives Input component should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "width": "80%",
      },
    ]
  }
  width="80%"
>
  <TextInput
    editable={true}
    onChangeText={[MockFunction]}
    placeholder="Placeholder"
    placeholderTextColor="#e5e5e5"
    style={
      Array [
        Object {
          "borderBottomColor": "#e5e5e5",
        },
        Object {
          "borderBottomWidth": 1,
          "fontSize": 16,
          "padding": 0,
          "paddingBottom": 5,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
